### PR TITLE
[release-4.15] MULTIARCH-4084: Reduce the policy access scope to specific instance

### PR DESCRIPTION
**Pull request by Neha-dot-Yadav**
This is a cherry-pick of #3491

### CODE REVIEW

File: cmd/infra/powervs/create.go [hunk 1]
```diff
@@ -388,28 +384,43 @@ func (infra *Infra) SetupInfra(ctx context.Context, options *CreateInfraOptions)
 		return fmt.Errorf("cloud connection is not up: %w", err)
 	}
 
+	if err = infra.setupSecrets(options); err != nil {
```
Comment (dharaneeshvrd) [inline cmd/infra/powervs/create.go:387]: Please comment, why it needs to be after the resources created

File: cmd/infra/powervs/destroy.go [hunk 2]
```diff
@@ -218,6 +213,11 @@ func (options *DestroyInfraOptions) DestroyInfra(ctx context.Context, infra *Inf
 		}
 	}
 
+	if err = deleteSecrets(options.Name, options.Namespace, powerVsCloudInstanceID, accountID, resourceGroupID); err != nil {
```
Comment (dharaneeshvrd) [inline cmd/infra/powervs/destroy.go:216]: Any reason to move this after deleteCOS?

File: cmd/infra/powervs/destroy.go [hunk 3]
```diff
@@ -302,14 +302,16 @@ func deleteDNSRecords(ctx context.Context, options *DestroyInfraOptions) error {
 }
 
 // deleteSecrets delete secrets generated for control plane components
-func deleteSecrets(name, namespace, accountID string, resourceGroupID string) error {
+func deleteSecrets(name, namespace, CloudInstanceID string, accountID string, resourceGroupID string) error {
```
Comment (dharaneeshvrd) [inline cmd/infra/powervs/destroy.go:305]:
```suggestion
func deleteSecrets(name, namespace, cloudInstanceID string, accountID string, resourceGroupID string) error {
```

File: cmd/infra/powervs/service_id.go [hunk 4]
```diff
@@ -193,6 +203,10 @@ func extractServiceIDFromCRN(crn string) string {
 	return crnL[len(crnL)-1]
 }
 
+func updateCRYaml(crYaml, serviceInstanceValue string) string {
+	return fmt.Sprintf(crYaml, serviceInstanceValue)
```
Comment (dharaneeshvrd) [inline cmd/infra/powervs/service_id.go:207]: Its good practice to use templating here.

**Review by dharaneeshvrd — CHANGES_REQUESTED** [review]

File: cmd/infra/powervs/create.go (hunk 1 above)
Comment (Neha-dot-Yadav) [reply cmd/infra/powervs/create.go:388]: Added a comment, 
please let me know if it is understandable?

File: cmd/infra/powervs/destroy.go (hunk 2 above)
Comment (Neha-dot-Yadav) [reply cmd/infra/powervs/destroy.go:216]: Yes, deleteSecrets needs `powerVsCloudInstanceID` value, which it is getting from the code after deleteCOS.

File: cmd/infra/powervs/service_id.go (hunk 4 above)
Comment (Neha-dot-Yadav) [reply cmd/infra/powervs/service_id.go]: Added the provision to use templating,
please let me know, if any further changes are required?

**Pull request closed without merging**